File: Warehousing.Data/Entities/Category.cs
namespace Warehousing.Data.Entities
{
    public class Category
    {
        public int Id { get; set; }
        public string? NameEn { get; set; } = string.Empty;
        public string NameAr { get; set; } = string.Empty;
        public string? Description { get; set; } = string.Empty;
        public bool IsActive { get; set; } = true;
        public virtual ICollection<SubCategory> SubCategories { get; set; } = new List<SubCategory>();
    }
}

File: Warehousing.UI/src/app/admin/models/category.ts
import { SubCategory } from "./SubCategory";

export interface Category {
    id: number;
    nameEn: string | null;
    nameAr: string;
    description: string | null;
    imagePath: string | null;
    isActive: boolean;
    createdAt: string | null;
    createdBy: string | null;
    updatedAt: string | null;
    updatedBy: string | null;
    subCategories: SubCategory[];
}